src/IPM/api.jl

```diff
@@ -0,0 +1,45 @@
+# API for getting internal members of IPM
+# The idea here is that a subtype of AbstractMadNLPSolver may specialize
+# some or all of these and the core functionality of the IPM in MadNLP
+# should function as normal.
+
+# getters for all fields
+for (k, attribute) in enumerate(fieldnames(MadNLPSolver))
+    fname = "_$(attribute)"
```

Comment:
I would suggest prefixing the getters with `get_`: 
```julia
fname = "get_$(attribute)"
```

Reply:
Done!